src/components/Modal.mock.ts

```diff
@@ -0,0 +1,29 @@
+import { FunctionComponent } from 'react'
+
+
+export type ModalExports = typeof import('src/components/Modal') & { __esModule: true }
```

Comment:
provided for Typescript consumers for added type safety when using with jest.mock().  See other comment below

https://github.com/DataBiosphere/terra-ui/pull/3497/files#diff-7d4292861b3947131d3928fff833cd84c19d84eae01bc3c90f467633d9129d0f

Reply:
The test could also define this type itself, right?

Especially since if the mocked module doesn't use a default export, there's no need for the ` & { __esModule: true }`.

Reply:
it could, yes.  The export is there as an option.  But it does require the `__esModule` prop on the typing.  We can't change the module exports contract from what the real module has.

Reply:
Jest transpiles ES modules to CommonJS and the `__esModule` flag makes the default export work nicely with CommonJS. If there's no default export, it's irrelevant.

> When using the factory parameter for an ES6 module with a default export, the __esModule: true property needs to be specified. This property is normally generated by Babel / TypeScript, but here it needs to be set manually. When importing a default export, it's an instruction to import the property named default from the export object:

https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options

Reply:
yes... the module being mocked does have a default export (which is the modal component fn we want to be mocked).  If you are talking generally, yes, if there is no default export, then `__esModule` workaround is not needed.

Reply:
Sorry, I should have been more clear. I was wondering, assuming a future with fewer default exports, whether we want to encourage this pattern of having the mock module export a type for the original module. Assuming no default exports, repeating `type ModalExports = typeof import('src/components/Modal')` seems more palatable than repeating `type ModalExports = typeof import('src/components/Modal') & { __esModule: true }`.